server/lib/get-data-for-token.js

```diff
@@ -56,63 +56,76 @@ module.exports = accessToken => {
     json: true
   }
 
-  return request(`${api.host}:${api.port}/api/client`, requestOptions)
-    .then(({results: clients}) => {
-      if (clients.length === 0) {
-        return Promise.reject()
+  const apiAddress =
+    api.serverAccessHost && api.serverAccessPort
+      ? `${api.serverAccessHost}:${api.serverAccessPort}`
+      : `${api.host}:${api.port}`
+
+  try {
+    const {results: clients} = await request(
+      `${apiAddress}/api/client`,
+      requestOptions
+    )
+
+    if (clients.length === 0) {
+      return Promise.reject()
+    }
+
+    response.client = clients[0]
+    response.config = config.get()
+
+    const languagesPromise = request(
+      `${apiAddress}/api/languages`,
+      requestOptions
+    )
+    const collectionsPromise = request(
+      `${apiAddress}/api/collections`,
+      requestOptions
+    )
+    const {results: languages} = await languagesPromise
+    const {collections} = await collectionsPromise
+
+    response.config.api.languages = languages
+
+    const {menu} = response.config.api
+
+    // A map corresponding collection names to a group, if they are part
+    // of one.
+    const collectionGroups = menu.reduce((groups, item) => {
+      const {collections, title} = item
+
+      if (Array.isArray(collections) && typeof title === 'string') {
+        collections.forEach(collection => {
+          groups.set(collection, title)
+        })
       }
 
-      response.client = clients[0]
-      response.config = config.get()
-
-      return request(`${api.host}:${api.port}/api/languages`, requestOptions)
-    })
-    .then(({results: languages}) => {
-      response.config.api.languages = languages
-
-      return request(`${api.host}:${api.port}/api/collections`, requestOptions)
-    })
-    .then(({collections}) => {
-      const {menu} = response.config.api
-
-      // A map corresponding collection names to a group, if they are part
-      // of one.
-      const collectionGroups = menu.reduce((groups, item) => {
-        const {collections, title} = item
-
-        if (Array.isArray(collections) && typeof title === 'string') {
-          collections.forEach(collection => {
-            groups.set(collection, title)
-          })
+      return groups
+    }, new Map())
+
+    // Augmenting collection objects with `_publishLink` properties, which
+    // contain a link to the collection (i.e. group + collection).
+    const augmentedCollections = collections
+      .map(collection => {
+        const group = collectionGroups.get(collection.slug)
+        const href = group
+          ? `/${slugify(group)}/${collection.slug}`
+          : `/${collection.slug}`
+
+        return {
+          ...collection,
+          fields: applyDefaultPublishParams(collection.fields),
+          _publishLink: href
```

Comment:
I just added two more properties to this response in #732. I think it'd be easier if we merged that one first and then you could rebase? 😇